### PR TITLE
New kvstore layer

### PR DESCRIPTION
**Pull request by tgraf**
Obsoleted by: #2708

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
The `07-kafka.sh` test keeps failing due to an error in resolving the identity of the kafka client. It only happens in the CI, not when run in my local environment.

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by manalibhutiyani** [comment]
@tgraf What is the motivation to allow a command line way to update (delete/set) Kvstore entries instead of just doing a get on the Key, value entries?

**Comment by tgraf** [comment]
> What is the motivation to allow a command line way to update (delete/set) Kvstore entries instead of just doing a get on the Key, value entries?

It gives up a possibility to remove eventual stale entries or manually reset the state in the kvstore.

**Comment by tgraf** [comment]
test-me-please

**Comment by raybejjani** [comment]
> It gives up a possibility to remove eventual stale entries or manually reset the state in the kvstore.

@tgraf Does this mean we could race with the agent when modifying certain keys? or does it know on which keys to enforce locking semantics (e.g. setting up a labels -> identity entry).

**Comment by tgraf** [comment]
> Does this mean we could race with the agent when modifying certain keys? or does it know on which keys to enforce locking semantics (e.g. setting up a labels -> identity entry).

The CLI tool does not know about locking. The agent must be resilient and capable of recovering from a manual interaction with the kvstore. Hence the controller which will, on a regular basis, sync every endpoint identity with the kvstore. The same can happen with the `cilium bpf` low level tool.

I will add a `(use caution)` warning to the CLI docs.

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

File: tests/00-script-linter.sh [hunk 1]
```diff
@@ -5,23 +5,6 @@ source "${dir}/helpers.bash"
 # dir might have been overwritten by helpers.bash
 dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
 
-function check_no_sleep {
```
Comment (nebril) [inline tests/00-script-linter.sh:8]: I liked this check :(. Can't we keep it?

Comment (tgraf) [reply tests/00-script-linter.sh:8]: It's just a workaround. It won't be merged. That said, the bash tests are being deprecated.

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
Obsoleted by #2708

**Pull request closed without merging**